test/unittests/eof_validation_test.cpp

```diff
@@ -1161,3 +1184,78 @@ TEST(eof_validation, jumpf_with_inputs_stack_overflow)
         EXPECT_EQ(validate_eof(code), EOFValidationError::stack_overflow);
     }
 }
+
+TEST(eof_validation, unreachable_code_sections)
+{
+    {
+        const auto code = eof_bytecode(OP_INVALID).code(OP_INVALID, 0, 0x80, 0);
+
+        EXPECT_EQ(validate_eof(code), EOFValidationError::unreachable_code_sections);
+    }
+
+    {
+        const auto code = eof_bytecode(bytecode{OP_CALLF} + "0001" + OP_STOP, 0)
+                              .code(bytecode{"5B"} + OP_RETF, 0, 0, 0)
+                              .code(bytecode{"FE"}, 0, 0x80, 0);
+
+        EXPECT_EQ(validate_eof(code), EOFValidationError::unreachable_code_sections);
+    }
+
+    {
+        const auto code = eof_bytecode(bytecode{OP_CALLF} + "0002" + OP_STOP, 0)
+                              .code(bytecode{"FE"}, 0, 0x80, 0)
+                              .code(bytecode{"5B"} + OP_RETF, 0, 0, 0);
+
+        EXPECT_EQ(validate_eof(code), EOFValidationError::unreachable_code_sections);
+    }
+
+    {
+        const auto code = eof_bytecode(bytecode{OP_CALLF} + "0003" + OP_STOP, 0)
+                              .code(bytecode{"FE"}, 0, 0x80, 0)
+                              .code(bytecode{"5B"} + OP_RETF, 0, 0, 0)
+                              .code(bytecode{OP_CALLF} + "0002" + OP_RETF, 0, 0, 0);
+
+        EXPECT_EQ(validate_eof(code), EOFValidationError::unreachable_code_sections);
+    }
+
+    {
+        const auto code =
+            eof_bytecode(bytecode{OP_JUMPF} + "0000").code(bytecode{OP_JUMPF} + "0001", 0, 0x80, 0);
+
+        EXPECT_EQ(validate_eof(code), EOFValidationError::unreachable_code_sections);
+    }
+
+    {
+        const auto code = eof_bytecode(bytecode{OP_JUMPF} + "0001")
+                              .code(bytecode{OP_STOP}, 0, 0x80, 0)
+                              .code(bytecode{"5B"} + OP_RETF, 0, 0, 0);
+
+        EXPECT_EQ(validate_eof(code), EOFValidationError::unreachable_code_sections);
+    }
+
+    {
+        std::string code_sections_256_err_001 = "E50001E50001";
+        std::string code_sections_256_err_254 = "E50001E50002";
+        std::string cs_calling_next;
+        for (int i = 0; i < 252; ++i)
+            cs_calling_next += "E5" + hex(big_endian(static_cast<uint16_t>(i + 1)));
```

Comment:
Not sure this does what you intended, maybe loop index is wrong, e.g. `code_sections_256_err_254` seems to end up with a sequence `jumpf{1} jumpf{2} jumpf{1} jumpf{2} jumpf{3} ...`
I think intention was to have `jumpf{1} jumpf{2} jumpf{3} jumpf{4} ...`

Also, if you want to use `bytecode` here, it should be possible with something like:
```cpp
auto code_sections_256_err_001 = eof_bytecode(bytecode{JUMPF} + "0001").code(bytecode{JUMPF} + "0002", 0, 0x80, 0);
for (int i = 2; i < 254; ++i)
  code_sections_256_err_001.code(bytecode{JUMPF} + big_endian(static_cast<uint16_t>(i + 1)), 0, 0x80, 0); 
```

(Also would be nice to have helpers in `bytecode.hpp` for callf / jumpf to not write `bytecode{JUMPF}` every time, but that's probably better for separate PR)

Reply:
There was an error in the loop indices, it is fixed and it is now using `bytecode`.